spark-jobs/src/main/scala/za/co/absa/enceladus/common/RecordIdGeneration.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common
+
+import com.typesafe.config.{Config, ConfigException}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.callUDF
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.enceladus.utils.udf.{UDFLibrary, UDFNames}
+
+object RecordIdGeneration {
+
+  sealed trait UuidType
+
+  object UuidType {
+    case object TrueUuids extends UuidType
+    case object PseudoUuids extends UuidType
+    case object NoUuids  extends UuidType
+  }
+
+  private val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  def getRecordIdGenerationStrategyFromConfig(conf: Config): UuidType = {
+    val strategyValue = conf.getString("enceladus.recordid.generation.strategy")
```

Comment:
**Question:** This makes the setting mandatory. Is it what we want?

Reply:
Yes, it does make it mandatory. This is in line with having the default value in the config.
I chose the combination deliberately. What would you prefer otherwise?

Reply:
Well, it's a pretty specific feature. And we require it to be set. I would make it optional and default to `None` obviously.

Reply:
Initially, I was under the impression since this is necessary for #1220, this feature would now be used by default. You are correct in the sense that it does not need to be, there just needs to be a dependency then (when attempting to send errors to Kafka, the ID Generation feature must be enabled. In other cases, it can be off).

Sure, I can change the default. Are we, though, sure what should be the default behavior? @benedeki @yruslan @Zejnilovic

Reply:
I prefer your initial implementation slightly more. Most of our config options are mandatory with defaults specified in `application.config.template`. And this case looks like a good candidate. And I'd set to `uuid` by default.

I see `enceladus_record_id` the same way as `enceladus_info_date`, etc. So they should be generated by default.

Reply:
Considering the rest of options, yeah mandatory is probably OK.
But then I would vote for `None` as the default in template.